### PR TITLE
Update RELEASE NOTES (breaking changes), README (link to release notes) and DOCUMENTATION

### PR DESCRIPTION
**Pull request by 64J0**
## Description

With this PR, I'm tackling the points raised at issue https://github.com/giraffe-fsharp/Giraffe/issues/606, on how to improve release notes regarding breaking changes.

## Related issues

- https://github.com/giraffe-fsharp/Giraffe/issues/606

### CODE REVIEW

**Comment by aspnetde** [comment]
Personally, I would give it its own section (maybe between videos and license), so it's a bit more obvious through its presence in the TOC on top. But having a dedicated document is a good thing anyway. 👍

File: README.md [hunk 1]
```diff
@@ -36,6 +36,9 @@ Read [this blog post on functional ASP.NET Core](https://dusted.codes/functional
 
 If you'd like to learn more about the motivation of this project please read my [blog post on functional ASP.NET Core](https://dusted.codes/functional-aspnet-core) (some code samples in this blog post might be outdated today).
 
+> [!TIP]
+> Check the [release notes](./RELEASE_NOTES.md) document to know how the system is evolving, and to get more information about breaking changes.
+
```
Comment (abelbraaksma) [inline README.md:41]: I've never seen `[!TIP]` syntax, but it renders nicely! It helps with this standing out, looks good!

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -12,6 +12,13 @@ Combination of the tags:
 
 ## 7.0.0-alpha-001 - 2024-07-08
 
+#### Breaking changes
+
+- `NewtonsoftJson` and `SystemTextJson` modules were removed by PR [Promote System.Text.Json as default JSON serializer](https://github.com/giraffe-fsharp/Giraffe/pull/563).
```
Comment (abelbraaksma) [inline RELEASE_NOTES.md:17]: Perhaps:

```suggestion
- `NewtonsoftJson` and `SystemTextJson` modules are removed, the latter is now module `Json`, NewtonSoft is not supported anymore, see PR [Promote System.Text.Json as default JSON serializer](https://github.com/giraffe-fsharp/Giraffe/pull/563).
```

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -12,6 +12,13 @@ Combination of the tags:
...
+    - Check the updated JSON documentation section to know more: [link](https://github.com/giraffe-fsharp/Giraffe/blob/master/DOCUMENTATION.md#json). Notice that there we have instructions on how to setup `NewtonsoftJson` back if you need.
```
Comment (abelbraaksma) [inline RELEASE_NOTES.md:18]: Maybe this?

```suggestion
    - For more information, check the updated JSON documentation section: [link](https://github.com/giraffe-fsharp/Giraffe/blob/master/DOCUMENTATION.md#json). This has instructions on how to setup `NewtonsoftJson` back if you need to keep it for your project.
```

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -12,6 +12,13 @@ Combination of the tags:
 
 ## 7.0.0-alpha-001 - 2024-07-08
 
+#### Breaking changes
```
Comment (abelbraaksma) [inline RELEASE_NOTES.md:15]: Perfect, make it stand out! :)

**Review by abelbraaksma — APPROVED** [review]
Thanks, this looks good. I have some suggestions, but nothing blocking.

File: README.md (hunk 1 above)
Comment (64J0) [reply README.md:41]: It was added by the end of last year, apparently. I discovered about it after reading this project from @mrtz-j: https://github.com/giraffe-fsharp/Giraffe.OpenApi.

Check this issue for more context on this feature discussion: https://github.com/orgs/community/discussions/16925.

**Pull request merged**